sys-utils/libcmt/Makefile

```diff
@@ -81,19 +74,62 @@ $(libcmt_SO): $(libcmt_OBJ)
 	$(TARGET_CC) -shared -o $@ $^
 
 libcmt: $(libcmt_LIB) $(libcmt_SO)
+install-run: $(libcmt_SO)
+	mkdir -p $(TARGET_DESTDIR)$(TARGET_PREFIX)/lib
+	cp -f $(libcmt_SO) $(TARGET_DESTDIR)$(TARGET_PREFIX)/lib
+
 install: $(libcmt_LIB) $(libcmt_SO) build/ffi.h
 	mkdir -p $(TARGET_DESTDIR)$(TARGET_PREFIX)/lib
-	cp -f $(libcmt_LIB) $(libcmt_SO) $(TARGET_DESTDIR)$(TARGET_PREFIX)/lib
+	cp -f $(libcmt_SO) $(libcmt_LIB) $(TARGET_DESTDIR)$(TARGET_PREFIX)/lib
 	mkdir -p $(TARGET_DESTDIR)$(TARGET_PREFIX)/include/libcmt/
 	cp -f include/libcmt/*.h $(TARGET_DESTDIR)$(TARGET_PREFIX)/include/libcmt/
 	cp -f build/ffi.h $(TARGET_DESTDIR)$(TARGET_PREFIX)/include/libcmt/
 	mkdir -p $(TARGET_DESTDIR)$(TARGET_PREFIX)/lib/pkgconfig
-	sed -e 's|@ARG_PREFIX@|$(TARGET_PREFIX)|g' src/libcmt.pc > $(TARGET_DESTDIR)$(TARGET_PREFIX)/lib/pkgconfig/libcmt.pc
-
-debian-package: install
-	mkdir -p $(TARGET_DESTDIR)/DEBIAN
-	sed 's|ARG_VERSION|$(LIBCMT_VERSION)|g;' tools/template/cross-control.template > $(TARGET_DESTDIR)/DEBIAN/control
-	dpkg-deb -Zxz --root-owner-group --build $(TARGET_DESTDIR) $(LIBCMT_DEB_FILENAME)
+	sed -e 's|@PREFIX@|$(TARGET_PREFIX)|g' \
+	    tools/libcmt.pc.in > $(TARGET_DESTDIR)$(TARGET_PREFIX)/lib/pkgconfig/libcmt.pc
+
+# requires either install-run or install
+debian-package:
+	mkdir -p $(dir $(DEB_FILENAME))
+	dpkg-deb -Zxz --root-owner-group --build $(TARGET_DESTDIR) $(DEB_FILENAME)
+
+$(TARGET_DESTDIR)/DEBIAN/control: tools/control.in
+	mkdir -p $(@D)
+	sed -e 's|@PACKAGE@|$(ARG_PACKAGE)|' \
+	    -e 's|@VERSION@|$(ARG_VERSION)|' \
+	    -e 's|@ARCHITECTURE@|$(ARG_ARCHITECTURE)|' \
+	    -e 's|@SECTION@|$(ARG_SECTION)|' \
+	    -e 's|@PROVIDES@|$(ARG_PROVIDES)|' \
+	    -e 's|@DESCRIPTION@|$(ARG_DESCRIPTION)|' \
+	    $< > $@
+
+# debian package variants
+libcmt-v$(ARG_VERSION).deb:
+	$(MAKE) $(TARGET_DESTDIR)/DEBIAN/control \
+	        ARG_PACKAGE=libcmt \
+	        ARG_ARCHITECTURE=riscv64 \
+	        ARG_PROVIDES=libcmt \
+	        ARG_SECTION=libs \
+	        ARG_DESCRIPTION="Cartesi Machine Tools"
+	$(MAKE) debian-package DEB_FILENAME=build/deb/$@
+
+libcmt-dev-v$(ARG_VERSION).deb:
```

Comment:
O downloaded the .deb artifacts from CI to test them, and both `libcmt-dev` and `libcmt` contains `/usr/lib/libcmt.so`, and they conflicts when installing both. Ideally `libcmt-dev` should not contain `libcmt.so`, so we can install both.

But if you remove, will we need a `libcmt-riscv64-cross` ? Because its `-dev` contains `libcmt.so`

Reply:
My idea with this was to only needing to install the `-dev` and leave the `run` only when "deploying", but to be honest I'm not too familiar with how this is usually done.